samples/bluetooth/direct_test_mode/src/fem/fem.c

```diff
@@ -155,3 +157,136 @@ uint32_t fem_radio_rx_ramp_up_delay_get(bool fast, nrf_radio_mode_t mode)
 
 	return ramp_up;
 }
+
+uint32_t fem_default_ramp_up_time_get(bool rx, nrf_radio_mode_t mode)
+{
+	bool fast_ramp_up = nrf_radio_modecnf0_ru_get(NRF_RADIO);
+
+	return rx ? fem_radio_rx_ramp_up_delay_get(fast_ramp_up, mode) :
+		    fem_radio_tx_ramp_up_delay_get(fast_ramp_up, mode);
+}
+
+int fem_tx_configure(uint32_t ramp_up_time)
+{
+	int32_t err;
+
+	fem_activate_event.event.timer.counter_period.end = ramp_up_time;
+
+	err = mpsl_fem_pa_configuration_set(&fem_activate_event, &fem_deactivate_evt);
+	if (err) {
+		printk("PA configuration set failed (err %d)\n", err);
+		return -EFAULT;
+	}
+
+	return 0;
+}
+
+int fem_rx_configure(uint32_t ramp_up_time)
+{
+	int32_t err;
+
+	fem_activate_event.event.timer.counter_period.end = ramp_up_time;
+
+	err = mpsl_fem_lna_configuration_set(&fem_activate_event, &fem_deactivate_evt);
+	if (err) {
+		printk("LNA configuration set failed (err %d)\n", err);
+		return -EFAULT;
+	}
+
+	return 0;
+}
+
+int fem_tx_gain_set(uint32_t gain)
+{
+	int32_t err;
+	mpsl_fem_gain_t fem_gain = { 0 };
+
+	/* Fallback to FEM specific function. It can be used for checking the valid output power
+	 * range.
+	 */
+	if (fem_api->tx_gain_set) {
```

Comment:
No more remarks from my side regarding the functional aspects, however the `tx_gain_set` function actually does not set the gain, istead it validates, that a specific gain is available for the given FEM.

Shouldn't it be renamed to something like `validate_gain`?